src/services/download/mod.rs

```diff
@@ -120,12 +147,29 @@ impl DownloadService {
         let hub = Hub::current();
 
         self.worker
-            .spawn("service.download", Duration::from_secs(3600), move || {
+            .spawn("service.download", Duration::from_secs(1800), move || {
                 dispatch_download(source, destination).bind_hub(hub)
             })
             // Map all SpawnError variants into DownloadErrorKind::Canceled.
             .map(|o| o.unwrap_or_else(|_| Err(DownloadErrorKind::Canceled.into())))
     }
+
+    pub fn list_files(
+        &self,
+        source: SourceConfig,
+        filetypes: &'static [FileType],
+        object_id: ObjectId,
+    ) -> impl std::future::Future<Output = Result<Vec<SourceFileId>, DownloadError>> {
+        let hub = Hub::current();
+        self.worker
+            .spawn(
+                "service.download.list_files",
+                std::time::Duration::from_secs(1800),
+                move || dispatch_list_files(source, filetypes, object_id).bind_hub(hub),
+            )
```

Comment:
It might be worth to avoid spawning for the ones which are implemented synchronously

Reply:
Agreed. Considering that we only have a single variant that needs a spawn, how would you feel about moving the `match` in here and then just spawn in the `SourceConfig::Sentry` arm? We can refactor this again if we choose to add more logic in the future.